plugins/balancedLevelProvider.go

```diff
@@ -93,6 +108,26 @@ func validateSpread(spread float64) {
 
 // GetLevels impl.
 func (p *balancedLevelProvider) GetLevels(maxAssetBase float64, maxAssetQuote float64) ([]api.Level, error) {
+	log.Printf("Last base balance was: %v", p.lastMaxAssetBase)
+	log.Printf("Last quote balance was: %v", p.lastMaxAssetQuote)
+
+	//Checking whether balances have changed meaningfully
+	//We have to give the balances some room to move because slightly different balance values come back from Horizon even if nothing changed
+	//Probably because floating point math is the worst
+	minValidBaseBalance := p.lastMaxAssetBase * 0.9999
+	maxValidBaseBalance := p.lastMaxAssetBase * 1.0001
+
+	minValidQuoteBalance := p.lastMaxAssetQuote * 0.9999
+	maxValidQuoteBalance := p.lastMaxAssetQuote * 1.0001
+
+	validBaseTrigger := maxAssetBase >= minValidBaseBalance && maxAssetBase <= maxValidBaseBalance
+	validQuoteTrigger := maxAssetQuote >= minValidQuoteBalance && maxAssetQuote <= maxValidQuoteBalance
+
+	if validBaseTrigger && validQuoteTrigger {
+		log.Println("Balances remain essentially the same as the previous cycle, leave levels as they are.")
+		return p.lastLevels, nil
+	}
+
```

Comment:
the code that is unchanged should be extracted into a new function:
`func (p *balancedLevelProvider) recomputeLevels(maxAssetBase float64, maxAssetQuote float64) ([]api.Level, error)` -- this will allow us to easily test it (whenever we get to that)

we can `return p.recomputeLevels(...)` after we set the 3 last values as you've done below